engine/src/main/battlecode/world/Island.java

```diff
@@ -92,11 +92,16 @@ public void advanceTurn() {
             return;
         }       
         int[] numOccupied = new int[2];
+        int roundNum = this.gw.getCurrentRound();
         for (MapLocation loc : this.locations) {
             InternalRobot robot = gw.getRobot(loc);
             if (robot != null) {
                 Team robotTeam = robot.getTeam();
                 numOccupied[robotTeam.ordinal()] ++;
+                if (roundNum%GameConstants.ISLAND_STRENGTH == 0 && robotTeam == this.teamOwning){
+                    int healingAmount = this.anchorPlanted == Anchor.STANDARD ? GameConstants.STANDARD_HEALING : GameConstants.ACCELERATING_HEALING;
```

Comment:
Do we want to do this here or after teamOwning is updated, I'm also down to do it here just for ease but I think we should document that